### ISSUE TITLE
[VersionControl] Add validations creating branches in Git

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7106]
There aren't validations creating branches, we can obtain errors depending on name length, etc.

```
ERROR [2019-02-08 10:51:42Z]: An unhandled exception has occurred. Terminating MonoDevelop? False
LibGit2Sharp.InvalidSpecificationException: invalid path for filesystem '/Users/jsuarezruiz/Projects/DefaultCollection/TipCalcCIGit/_git/TipCalcCIGit/.git/refs/heads/dsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uydsfsdfdsdsfert245906u430rieetñyu+çilñop+ç09+98op7uy': File name too long
  at LibGit2Sharp.Core.Ensure.HandleError (System.Int32 result) [0x00064] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Ensure.cs:155
  at LibGit2Sharp.Core.Ensure.ZeroResult (System.Int32 result) [0x0000c] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Ensure.cs:173
  at LibGit2Sharp.Core.Proxy.git_branch_create_from_annotated (LibGit2Sharp.Core.Handles.RepositoryHandle repo, System.String branch_name, System.String targetIdentifier, System.Boolean force) [0x00020] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Proxy.cs:195
  at LibGit2Sharp.BranchCollection.Add (System.String name, System.String committish, System.Boolean allowOverwrite) [0x00019] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/BranchCollection.cs:158
  at LibGit2Sharp.BranchCollection.Add (System.String name, LibGit2Sharp.Commit commit, System.Boolean allowOverwrite) [0x0000d] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/BranchCollection.cs:143
  at LibGit2Sharp.BranchCollection.Add (System.String name, LibGit2Sharp.Commit commit) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/BranchCollection.cs:129
  at LibGit2Sharp.RepositoryExtensions.CreateBranch (LibGit2Sharp.IRepository repository, System.String branchName, LibGit2Sharp.Commit target) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/RepositoryExtensions.cs:140
  at MonoDevelop.VersionControl.Git.GitRepository.CreateBranchFromCommit (System.String name, LibGit2Sharp.Commit id) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1225
  at MonoDevelop.VersionControl.Git.StashManagerDialog.OnButtonBranchClicked (System.Object sender, System.EventArgs e) [0x0008f] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/StashManagerDialog.cs:143
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0 (System.Object state) [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1023
  at MonoDevelop.Ide.DispatchService+GtkSynchronizationContext+TimeoutProxy.HandlerInternal (System.IntPtr data) [0x00016] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/DispatchService.cs:77
Exception Data:
libgit2.code: -12
libgit2.category: -1
```

> VS bug [#789303](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789303)

**Comment by jsuarezruiz** [comment]
Related to https://github.com/mono/monodevelop/issues/4406

**Comment by jsuarezruiz** [comment]
Sent PR: https://github.com/mono/monodevelop/pull/7191

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1624 (master)

Pull Request #7191 merged by: Vsevolod Kukol
Author: mono
Commit: 3ea7c84a26baa63ac2f429d531d59d9bde71a81e (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2394 (release-8.0)

Pull Request #7191 merged by: Vsevolod Kukol
Author: mono
Commit: 3ea7c84a26baa63ac2f429d531d59d9bde71a81e (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7191 merged by: Vsevolod Kukol
Author: mono
Commit: 3ea7c84a26baa63ac2f429d531d59d9bde71a81e (xamarin/monodevelop)
